### DISCUSSION TITLE
質問回答

### DISCUSSION THREAD

**Discussion by kenton116** [satsukizzz/n-playspace-http#21] [Q&A]
質問をしたりそれに答えたり

**Comment by kenton116** [comment]
cloneしたリポジトリ（つまりローカル）でブランチを作るのはどうするのですか？

**Reply by kenton116** [reply]
それをプッシュする方法も知らないのでお願いします

**Reply by satsukizzz** [reply]
なるほど、授業ではやっていませんね

`git checkout -b new-branch-name` でnew-branch-nameという名前のブランチを新しく作りcheckoutします。 `-b` というオプションがあると新しくブランチを作るんですね。

**Reply by kenton116** [reply]
なるほど！

**Reply by kenton116** [reply]
ありがとうございます！
リーダー流石です！

**Comment by HappyKumamoto** [comment]
質問コーナー作成ありがとうございます。
いまだにbranchのところで行き詰まっております。
git pull の後、git branch で見ると、main と first-env だけなので
git branch -a としてみると、こういうふうに表示されます。
どうすれば良いでしょうか。どうかご指導お願いいたします。
![211021](https://user-images.githubusercontent.com/67646107/138226579-f95655c9-6910-40e3-8bbe-10690d4560b6.png)

**Reply by satsukizzz** [reply]
何をしたいのか、教えていただけますか？
現状はその挙動で問題がないです。
ブランチを作りたいのか、誰かの既存のブランチに入りたいのか、それにより確認することが変わってきます。

**Comment by HappyKumamoto** [comment]
プルリクをあげてもらっているので、動作確認してレビューするために、そのブランチに入りたいのですが・・・
ここからの入り方が分からず、困っています。
本当に初歩的すぎて申し訳ありません。🙇

**Reply by satsukizzz** [reply]
https://github.com/satsukizzz/n-playspace-http/pull/23
こちらですね！気づいてなかったです 💦 
マージ元が `nyeoro-bong:nyoro_314` とあるので、コマンドだけ挙げるとこうなります

```
git remote add nyeoro-bong git@github.com:nyeoro-bong/n-playspace-http.git
git fetch nyeoro-bong
git checkout nyeoro-bong/nyoro_314
```

すこし解説すると、
1. にょろぼんさんのフォークしたリポジトリをremoteとして追加して、fetchします
1. その後、プルリクエストが出ているブランチにcheckoutします。

といった感じです！

**Reply by HappyKumamoto** [reply]
詳しいご説明、ありがとうございます。

ブランチはシンプルに main から枝分かれしているイメージしか持っていなかったもので、
まだまだ勉強が足りないようです。🙇

**Reply by satsukizzz** [reply]
ぐちゃぐちゃに…なりますよ…… 😨 
シンプルにするためにも"ブランチポリシー"というものを設けたりします。
ただ、ポリシー(=ルール)を作りすぎるとメンドクサイですから、まずはルールが欲しくなるまでいっぱいやってみようという魂胆です。 🤚
